documentation/internal/Element-Type-Specs.md

```diff
@@ -0,0 +1,129 @@
+  - default: `{"value": true}`
+- **validation_message**: `string` -- the message that shows in the UI when validation fails.
+
+## Question/Element types
+
+**Note**: all parameter fields can also have a dynamic query object instead of a primitive. The `evaluateExpression` function will return literal strings (or numbers, booleans) as is. The types described for the parameters below are the type that is expected to be _returned_ from a query expression.
+
+### Short Text Input _(not yet implemented -- highest priority)_
+
+- **type/code**: `shortText`
+- **category**: `Question`
+
+_Free-form, single-line text input element_
+
+#### Input parameters (in the `parameters` JSON)
+
+- **label**: `string` -- Text that shows in the HTML "label" attribute of the form element (suggestion: make this interpret Markdown)
+- **description**: `string` -- additional explanatory text (usually not required) [Optional]
+- **placeholder**: `string`-- text to display before user input (HTML "placeholder" attribute) [Optional]
+- **maskedInput**: `boolean` -- if `true`, displays user input as masked (hidden) characters -- i.e. for passwords. [Optional]
+- **minWidth**/**maxWidth**: `integer` -- optional controls over visual display [Optional] _(We may never use these)_
+
+#### Response type
+
+_This describes the expected object that will be stored in the `application_response` table `value` field. from the user's response_
+`{ text: <string> }`
+
+---
+
+### Text Information _(not yet implemented)_
+
+- **type/code**: `textInfo`
+- **category**: `Information`
+
+_For displaying blocks of text in the application_
+
+#### Input parameters
+
+- **title**: `string` -- Heading text to display [Optional]
+- **text**: `string` -- body text to display
+  _(Maybe have some formatting options, but not initially, although `text` should support Markdown)_
+
+---
+
+### Radio Buttons _(not yet implemented)_
+
+- **type/code**: `radioChoice`
+- **category**: `Question`
+
+_Multi-choice question, with one allowed selection, displayed as labelled radio buttons_
+
+#### Input parameters
+
+- **label**: `string` -- as above
+- **description**: `string` -- as above [Optional]
+- **options**: `array[string]` -- the options for the radio buttons
+- **default**: `string`/`number` -- the value initially selected before user input. If `number`, refers to the index of the options array. If not provided, no options will be pre-selected.
+- **hasOther**: `boolean` -- if `true`, an additional text-entry field is provided so the user can add their own alternative option _(may not implement in first iteration but good to have the option in future)_
+
+#### Response type
+
+```
+{
+  optionIndex: <integer> (index from the options array)
+  text: <string> (actual text from options array)
+  reference?: To-do: some way to link the response to an entity in database (e.g. an organisation)
+  -- @nmadruga maybe you have an idea of what we'd need here
```

Comment:
I think to be able to have this as part of the response the **Input** would also need to receive that information... In my thinking the options would receive not only a string per row but what some reference to what it is linked to. But maybe using a different element plugin - e.g. Radio Buttons Link ?

Reply:
Let's discuss soon, but will probably be important when we do #90, etc.